colorbleed/widgets/popup.py

```diff
@@ -104,6 +104,107 @@ def calculate_window_geometry(self):
         return QtCore.QRect(x, y, width, height)
 
 
+class Popup2(QtWidgets.QDialog):
+
+    on_show = QtCore.Signal()
+
+    def __init__(self, parent=None, *args, **kwargs):
+        QtWidgets.QDialog.__init__(self, parent=parent, *args, **kwargs)
+        self.setContentsMargins(0, 0, 0, 0)
+
+        # Layout
+        layout = QtWidgets.QHBoxLayout(self)
+        layout.setContentsMargins(10, 5, 10, 10)
+        message = QtWidgets.QLabel("")
+        message.setStyleSheet("""
+        QLabel {
+            font-size: 12px;
+        }
+        """)
+        show = QtWidgets.QPushButton("Fix")
+        show.setSizePolicy(QtWidgets.QSizePolicy.Maximum,
+                           QtWidgets.QSizePolicy.Maximum)
+        show.setStyleSheet("""QPushButton { background-color: #BB0000 }""")
+
+        toggle = QtWidgets.QCheckBox()
+        toggle.setText("Update Keys")
+
+        layout.addWidget(message, 1)  # Ensure stretch
+        layout.addWidget(toggle)
+        layout.addWidget(show)
+
+        # Size
+        self.resize(400, 40)
+        geometry = self.calculate_window_geometry()
+        self.setGeometry(geometry)
+
+        self.widgets = {
+            "message": message,
+            "show": show,
+            "toggle": toggle,
+        }
+
+        # Signals
+        show.clicked.connect(self._on_show_clicked)
+
+        # Set default title
+        self.setWindowTitle("Popup")
+
+    def setMessage(self, message):
+        self.widgets['message'].setText(message)
+
+    def _on_show_clicked(self):
+        """Callback for when the 'show' button is clicked.
+
+        Raises the parent (if any)
+
+        """
+
+        parent = self.parent()
+        self.close()
+
+        # Trigger the signal
+        self.on_show.emit()
+
+        if parent:
+            parent.raise_()
+
+    def calculate_window_geometry(self):
```

Comment:
Seem like code duplication? Does it make more sense to make a base Popup class that you can inherit from?